### PR TITLE
Adding README.macOS.arm.bash for MAC arm based OS for 6X_stable

### PR DESCRIPTION
**Pull request by Shakarada**
Authored-by: Anusha Shakarad [shakarada@vmware.com](mailto:shakarada@vmware.com)
Co-authored-by: Mounika M [mmounika@vmware.com](mailto:mmounika@vmware.com)

Problem :
With the existing README.macOS.bash script there were multiple dependencies missing while configuring gpdb on M1 based MacOS. The missing dependencies resulted in random errors and fixing these were time consuming.

RCA :
Missing dependencies were due to change in architecture from x86 to arm64 .

Solution :
Below are the dependencies needed for arm based macOS which are implemented in the README.macOS.arm.bash script

Change the brew installation method from ruby to bash and update the homebrew path.
library xerces-c installation method changed from brew installation to manual make installation.
Hardcoded the python installation to 2.7.18 & 3.9 version specific to gpdb compilation
Updated bashrc PATH variables specific to gpdb
Arm based MacOS users can now use README.macOS.arm.bash script for their macOS setup instead of README.macOS.bash script

### CODE REVIEW

File: README.macOS.arm.bash [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+rm -rf /opt/homebrew/bin/python3
+rm -rf /opt/homebrew/bin/pip3
+brew install python@3.9
+python_version=$(echo `ls  /opt/homebrew/Cellar/python@3.9/`)
+ln -s /opt/homebrew/Cellar/python@3.9/$python_version/bin/python3.9 /opt/homebrew/bin/python3
+ln -s /opt/homebrew/Cellar/python@3.9/$python_version/bin/pip3.9 /opt/homebrew/bin/pip3
+pip3 install -r python-dependencies.txt
+pyenv install $python_version
+
+#echo -e "127.0.0.1\t$(hostname)" | sudo tee -a /etc/hosts
```
Comment (shrakesh) [inline README.macOS.arm.bash:75]: is it intentional, can we remove this commented line?

**Review by shrakesh — APPROVED** [review]
approving with the following minor comments. 
1.  update the Python version on pr description. 
2. clean unwanted commented portion of the script.

File: README.macOS.arm.bash (hunk 1 above)
Comment (Shakarada) [reply README.macOS.arm.bash]: Removed thanks

**Comment by Shakarada** [comment]
> approving with the following minor comments.
> 
> 1. update the Python version on pr description.
> 2. clean unwanted commented portion of the script.

**Review by ravoorsh — APPROVED** [review]
LGTM

**Pull request merged**